### ISSUE TITLE
mistune improvement

### ISSUE THREAD

**Issue by lepture** [nitely/Spirit#53]
Hi, I am the author of mistune. I found those `audio_link`, `image_link` etc in BlockLexer. Is there any particular reason that they should be in a lexer?

I think it would be a simple renderer like this:

```python
class LinkRenderer(mistune.Renderer):
    def autolink(self, link, is_email=False):
        if is_email:
            return 'email link'
        if is_youtube(link):
            return 'youtube html'
        if is_audio(link):
            return 'audio html'
        ...
```

**Comment by nitely** [comment]
HI!

First, let me thank you for such an awesome library, good job!

I did write a (renderer)[https://github.com/nitely/Spirit/blob/master/spirit/utils/markdown/renderer.py], but honestly I don't remember anything about all the custom changes I made, since it was a long ago. I'm guessing that's the way you did it in mistune 0.3 which is the version I'm currently using (yes, I should update it, sorry about that) since I usually read the code. Maybe there's a better way to achieve the same in a more recent version.

I'd argue your autolink function is doing too much, but that's probably for the sake of brevity and you are trying to show me something I don't get just yet :smile:
